packages/credential-data-integrity/src/suites/VCkitEddsaRdfc2022.ts

```diff
@@ -0,0 +1,164 @@
+import {
+  bytesToBase64,
+  hexToBytes,
+  base64ToBytes,
+  bytesToBase58,
+} from '@veramo/utils';
+import { VeramoLdSignature } from '@veramo/credential-ld';
+import {
+  CredentialPayload,
+  IAgentContext,
+  DIDDocument,
+  DIDDocComponent,
+  IKey,
+  TKeyType,
+  IResolver,
+  IKeyManager,
+} from '@veramo/core-types';
+import * as Ed25519Multikey from '@digitalbazaar/ed25519-multikey';
+import { DataIntegrityProof } from '@digitalbazaar/data-integrity';
+import { cryptosuite as eddsaRdfc2022CryptoSuite } from '@digitalbazaar/eddsa-rdfc-2022-cryptosuite';
+
+/**
+ * Veramo wrapper for the Ed25519Multikey suite by @digitalbazaar
+ */
+export class VCkitEddsaRdfc2022 extends VeramoLdSignature {
+  /**
+   * This method is called when create a suite's verification key.
+   * @returns the supported verification type (Multikey)
+   */
+  getSupportedVerificationType(): string {
+    return 'Multikey';
+  }
+
+  /**
+   * This method is called in the custom signer object in the suite.
+   * @returns the supported Veramo key type (Ed25519)
+   */
+  getSupportedVeramoKeyType(): TKeyType {
+    return 'Ed25519';
+  }
+
+  /**
+   * This method is called before the credential is signed.
+   * If you need to modify the suite before signing, you can do it here.
+   * @param key is the Cryptographic key to be used for signing
+   * @param issuerDid is the DID of the issuer
+   * @param verificationMethodId is the ID of the verification method
+   * @param context is the agent context
+   * @returns the suite for signing
+   */
+  async getSuiteForSigning(
+    key: IKey,
+    issuerDid: string,
+    verificationMethodId: string,
+    context: IAgentContext<IResolver & Pick<IKeyManager, 'keyManagerSign'>>,
+  ) {
+    // Custom signer for the suite
+    const signer = {
+      // This field required for the DataIntegrityProof suite
+      id: verificationMethodId,
+      // This field required for the DataIntegrityProof suite
+      algorithm: this.getSupportedVeramoKeyType(), // Ed25519
+      sign: async (args: { data: Uint8Array }): Promise<Uint8Array> => {
+        const messageString = bytesToBase64(args.data);
+        const signature = await context.agent.keyManagerSign({
+          keyRef: key.kid,
+          data: messageString,
+          encoding: 'base64',
+          algorithm: 'EdDSA',
+        });
+        return base64ToBytes(signature);
+      },
+    };
+
+    const verificationKey = await Ed25519Multikey.from({
+      id: verificationMethodId,
+      controller: issuerDid,
+      type: this.getSupportedVerificationType(), // Multikey
+      publicKeyMultibase: `z${bytesToBase58(hexToBytes(key.publicKeyHex))}`,
```

Comment:
Can you provide the reference link to make it clear?

Reply:
Yep, I'll update it.